File: Algorithms/YandexTraining/1/5/1.cpp
#include <iostream>
#include <vector>
#include <utility>
#include <cstdlib> 

int main() {
    int n;
    std::cin >> n;
    std::vector<int> shirts(n);
    for (auto i = 0uz; i < n; ++i) std::cin >> shirts[i];

    int m;
    std::cin >> m;
    std::vector<int> pants(m);
    for (auto i = 0uz; i < m; ++i) std::cin >> pants[i];

    auto i = 0uz, j = 0uz;
    auto mindiff = std::abs(shirts[0] - pants[0]);
    std::pair<int,int> ans{shirts[0], pants[0]};

    while (i < shirts.size() && j < pants.size()) {
        auto a = shirts[i];
        auto b = pants[j];
        auto diff = std::abs(a - b);

        if (diff < mindiff) {
            mindiff = diff;
            ans = {a, b};
        }

        if (a < b) i++;
        else j++;
    }

    std::cout << ans.first << " " << ans.second << "\n";
    return 0;
}

File: Algorithms/YandexTraining/1/5/1.py
n = int(input())
shirts = list(map(int, input().split())) 
m = int(input())
pants = list(map(int, input().split()))

i = 0
j = 0

mindiff = abs(shirts[0] - pants[0])
ans = (shirts[0], pants[0])

while i < len(shirts) and j < len(pants):

    a = shirts[i]
    b = pants[j]

    diff = abs(a - b)
    if diff < mindiff:
        mindiff = diff
        ans = (a, b)

    if a < b:
        i += 1
    else:
        j += 1

print(*ans)